### PR TITLE
Adding different variations of empty and missing body specification tests

### PR DESCRIPTION
**Pull request by neilcampbell**
Can you please check these tests pass on the Ruby impl? Also they should be good to merge into the version-2.

### CODE REVIEW

**Comment by uglyog** [comment]
I've reviewed this and it looks good to me. I do have a conceptual issue in understanding what it means to say "a null body" with respect to HTTP interactions. I can understand an empty body, or a missing body. But I do agree that documenting what we expect when an implementation sets the body to a null value is important.

**Comment by neilcampbell** [comment]
Yeah I agree, I had the same conceptual argument in my head too. Will change the terminology to empty body.

**Comment by neilcampbell** [comment]
@uglyog Updated.

File: testcases/response/body/missing body found when empty expected.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "match": true,
+  "comment": "Missing body found, when an empty body was expected",
+  "expected" : {
+    "body": null
+  },
+  "actual": {
+  }
+}
```
Comment (bethesque) [inline testcases/response/body/missing body found when empty expected.json]: How could the actual not have a status code?

Comment (neilcampbell) [inline testcases/response/body/missing body found when empty expected.json]: @bethesque The real actual request wouldn't. You could make that argument with all the body tests in this repo (non of them have status code  specified). I assume we are doing this as the focus of the body tests is on body and not the status code.

**Pull request merged**